src/components/application_manager/include/application_manager/commands/command_request_impl.h

```diff
@@ -112,8 +112,15 @@ class CommandRequestImpl : public CommandImpl,
  public:
   enum RequestState { kAwaitingHMIResponse = 0, kTimedOut, kCompleted };
 
+  /**
+   * @brief The HashUpdateMode enum defines whether request has to update
+   * hash after its execution is finished
+   */
+  enum HashUpdateMode { kDoHashUpdate, kNoHashUpdate };
+
   CommandRequestImpl(const MessageSharedPtr& message,
-                     ApplicationManager& application_manager);
+                     ApplicationManager& application_manager,
+                     HashUpdateMode hash_update_mode = kNoHashUpdate);
```

Comment:
@dev-gh I can propose you to create some separate function like `CommandRequestImpl::SetHashUpdateMode()` and call it when it necessary to avoid this extra parameter with default value in ctor. And in initialization list: `hash_update_mode_(kNoHashUpdate)`